src/client.ts

```diff
@@ -8002,29 +8015,38 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * Make a request for an `m.login.token` to be issued as per
      * [MSC3882](https://github.com/matrix-org/matrix-spec-proposals/pull/3882).
      * The server may require User-Interactive auth.
-     * Note that this is UNSTABLE and subject to breaking changes without notice.
+     *
+     * Compatibility with unstable implementations of MSC3882 is deprecated and will be removed in a future release.
+     *
      * @param auth - Optional. Auth data to supply for User-Interactive auth.
      * @returns Promise which resolves: On success, the token response
      * or UIA auth data.
      */
     public async requestLoginToken(auth?: AuthDict): Promise<UIAResponse<LoginTokenPostResponse>> {
         // use capabilities to determine which revision of the MSC is being used
         const capabilities = await this.getCapabilities();
-        // use r1 endpoint if capability is exposed otherwise use old r0 endpoint
-        const endpoint = UNSTABLE_MSC3882_CAPABILITY.findIn(capabilities)
-            ? "/org.matrix.msc3882/login/get_token" // r1 endpoint
-            : "/org.matrix.msc3882/login/token"; // r0 endpoint
+
+        const usesStable = !!capabilities["m.get_login_token"];
+        const usesUnstableRevision1 = !usesStable && !!capabilities["org.matrix.msc3882.get_login_token"];
+        // use stable endpoint if capability is exposed otherwise use old unstable r0 endpoint
+        const endpoint =
+            usesStable || usesUnstableRevision1
+                ? "/login/get_token" // endpoint for stable + unstable r1
+                : "/login/token"; // unstable r0 endpoint
+
+        // determine whether to use stable or unstable prefix
+        const prefix = usesStable ? ClientPrefix.V1 : `${ClientPrefix.Unstable}/org.matrix.msc3882`;
```

Comment:
If I understand correctly, there are three different potential endpoints here? 

tbh I think it would be easier to set `prefix = ""` and embed the prefix in `endpoint`:

```suggestion
        let endpoint: String;
        if (capabilities["m.get_login_token"]) {
            // use the stable endpoint
            endpoint = ClientPrefix.V1 + "/login/get_token";
        } else if (capabilities["org.matrix.msc3882.get_login_token"]) {
            // newer unstable endpoint
            endpoint = ClientPrefix.Unstable + "/org.matrix.msc3882/login/get_token";
        } else {
            // old unstable endpoint
            endpoint = ClientPrefix.Unstable + "/org.matrix.msc3882/login/token";
        }
```

Reply:
(I also think that `ClientPrefix.V1` is an obscure way to write `"/_matrix/client/v1"`, but that's a side issue and I may be alone on this)

Reply:
Oh also, rather than hardcoding the capability names everywhere, let's use the symbolic constant, `GET_LOGIN_TOKEN_CAPABILITY.name` and `GET_LOGIN_TOKEN_CAPABILITY.altName`.

Reply:
> If I understand correctly, there are three different potential endpoints here?

Yes. And your proposal is much clearer to read so I have incorporated it. 👍